### DISCUSSION TITLE
How to specify different kinds of pay expr in the same parallel reduce (non network token and algo)

### DISCUSSION THREAD

**Discussion by alcyoneus-g** [reach-sh/reach-lang#1134] [Q&A] [answered]
```
  .api(
      BuyerApi.touch,
      () => {},
      () => 0,
      (k) => {
        k(null);
        return [
            balance(token),  
        ];
      }
    )
    .api(
      BuyerApi.reveal,
      () => {
        assume(lastConsensusSecs() > end, "reveal time hasn't been reached")
      },
      () => [[1, token]],
      (k) => {
```

I keep getting this issue:
```
reachc: error[RE0088]: These types are mismatched:
  expected: UInt
       got: Tuple(Tuple(UInt, Token))

  ./index.rsh:179:73:block app
```

there are some APIs that require algo payments and some require a token transfer, how to handle those without this issue?

**Comment by jeapostrophe** [comment, marked as answer]
This is what `.paySpec` is for --- https://docs.reach.sh/rsh/step/#p_32 --- you need to include everything everywhere and Reach will optimize away things that are 0

**Reply by alcyoneus-g** [reply]
thanks for the reply @jeapostrophe. How exactly do I add use that inside the parallel reduce loop? in the snippet I have shared?

**Reply by alcyoneus-g** [reply]
or any similar example snippet I could take as a reference, would be appreciated

**Reply by jeapostrophe** [reply]
`.paySpec([token])` and then you change the first to `() => [0, [0, token]]` and the second to `() => [0, [1, token]]`

**Reply by alcyoneus-g** [reply]
@jeapostrophe but what I want to achieve is that some APIs use ALGO as payment (that's why the first api is just 0 instead of tuple of tuple), and some api use the tokens like the second one.

Something like:
```
 .api(
      BuyerApi.api1,
      () => {},
      () => 1000000,
      (k) => {
        k(null);
        return [
        ];
      }
    )
    .api(
      BuyerApi.api2
      () => {},
      () => [[1, token]],
      (k) => {
```

**Reply by jeapostrophe** [reply]
That's what my solution would let you do.

**Reply by alcyoneus-g** [reply]
so, I get how it would work for 0  but how would I go about expressing `() => 1000000,` ? 

`() => [100000, [0, token]]` is correct?

**Reply by jeapostrophe** [reply]
Yup

**Reply by alcyoneus-g** [reply]
thanks for the help @jeapostrophe, it's has worked 👍
